src/topology/multicodec-topology.js

```diff
@@ -97,6 +99,25 @@ class MulticodecTopology extends Topology {
       }
     }
   }
+
+  /**
+   * Verify if a new connected peer has a topology multicodec and call _onConnect.
+   * @param {Connection} connection
+   * @returns {void}
+   */
+  _onPeerConnect (connection) {
+    const peerId = connection.remotePeer
+    const protocols = this._registrar.peerStore.protoBook.get(peerId)
+
+    if (!protocols) {
+      return
+    }
+
+    if (this.multicodecs.filter(multicodec => protocols.includes(multicodec)).length) {
```

Comment:
nit: I'd change this to a find, we don't need to go through all the multicodecs if we find one.